include/revng/LocalVariables/LocalVariableBuilder.h

```diff
@@ -276,7 +276,7 @@ class LocalVariableBuilder {
   ///
   /// TODO: This method can be made const whenever we drop legacy mode.
   AssignType *createAssignmentBefore(llvm::Value *LocationToAssign,
-                                     llvm::Value *ValueToAssign,
```

Comment:
Don't do this. The fact that we're not using this with anything but `Instruction`s today doesn't mean that we are not going to need a more general `Value` in the future.

Reply:
Then what would you think about an additional `DebugLoc` argument?

Reply:
That's probably the most general thing ever that you could do, but in practice I really think we only ever want to use the debug location from `ValueToAssign` if it's an instruction. So leave the prototype as it is, and just fix the body to use the debug info conditionally, when `ValueToAssign` is an `Instruction`.